backend/fms_core/models/sample.py

```diff
@@ -157,7 +164,9 @@ class Sample(TrackedModel):
     # TODO: I would prefer consistent terminology with Container if possible for this heirarchy
     container = models.ForeignKey(Container, on_delete=models.PROTECT, related_name="samples",
                                   limit_choices_to={"kind__in": SAMPLE_CONTAINER_KINDS},
-                                  help_text="Designated location of the sample.")
+                                  help_text="Designated location of the sample.",
+                                  error_messages={'null': 'Container could not be processed due to missing information. Kind, Name and Barcode are necessary.'})
```

Comment:
At this level, you only know that a container is missing ... This is too much to assume we are in sample submission and that the user did forget to fill in those fields. You could use this error message by inserting it manually at the level of the ressource (maybe on  the generic ressource since this could apply to multiple user submitted templates).

Reply:
Perfect. Should I just leave "Container could not be processed due to missing information"?
```suggestion
                                  error_messages={'null': 'Container could not be processed due to missing information.'})
```

Reply:
This implies again a user input failure. Since you are at the model level, i would go with something more agnostic like "Sample container cannot be null" which is of very little use to an end user. I would encourage you to work at a more context specific level to be able to provide clearer and more detailed error messages. The field definition will not be able to differentiate whether you create the object from a template submission or programmatically somewhere else. You should work within the ressource for that or alternatively in a module that will be informed of the context somehow.